WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSection.cs

```diff
@@ -0,0 +1,129 @@
+{
+    public class WiseSection : BaseBinaryDeserializer<WiseSectionHeader>
+    {
+        /// <inheritdoc/>
+        public override WiseSectionHeader? Deserialize(Stream? data)
+        {
+            // If the data is invalid
+            if (data == null || !data.CanRead)
+                return null;
+
+            // Cache the current offset
+            long initialOffset = data.Position;
+            try
+            {
+                var wiseSectionHeader = ParseWiseSectionHeader(data);
+
+                // Checks if version was able to be read
+                if (wiseSectionHeader.Version == null)
+                    return null;
+
+                // Main MSI file
+                if (wiseSectionHeader.MSIFileEntryLength == 0)
+                    return null;
+                else if (wiseSectionHeader.MSIFileEntryLength >= data.Length)
+                    return null;
+
+                // First executable file
+                if (wiseSectionHeader.FirstExecutableFileEntryLength == 0)
+                    return null;
+                else if (wiseSectionHeader.FirstExecutableFileEntryLength >= data.Length)
+                    return null;
+                
+                // Second executable file
+                if (wiseSectionHeader.SecondExecutableFileEntryLength == 0)
+                    return null;
+                else if (wiseSectionHeader.SecondExecutableFileEntryLength >= data.Length)
+                    return null;
+
+                return wiseSectionHeader;
+            }
+            catch
+            {
+                // Ignore the actual error
+                return null;
+            }
+        }
+        
+        
+        
+        /// <summary>
+        /// Parse a Stream into a WiseSectionHeader
+        /// </summary>
+        /// <param name="data">Stream to parse</param>
+        /// <returns>Filled WiseSectionHeader on success, null on error</returns>
+        private static WiseSectionHeader ParseWiseSectionHeader(Stream data)
+        {
+            var header = new WiseSectionHeader();
+            
+            int headerLength = -1; // indexed from 0
+            
+            // Find offset of "WIS", determine header length, read presumed version value
+            foreach (int offset in WisOffsets)
+            {
+                data.Seek(offset, 0);
+                byte[] checkBytes = data.ReadBytes(3);
+                if (checkBytes.EqualsExactly(WisString))
+                {
+                    headerLength = WiseSectionHeaderLengthDictionary[offset];
+                    data.Seek(offset - WiseSectionVersionOffsetDictionary[offset], 0);
+                    header.Version = data.ReadBytes(WiseSectionVersionOffsetDictionary[offset]);
+                    break;
+                }
+            }
+            
+            data.Seek(0, 0);
+            
+            // Is there a better way to handle returning at the proper time besides just hardcoding return checks?
```

Comment:
Is this comment relevant anymore?

Reply:
Yes; it feels weird to hardcode a bunch of if == statements that return, considering I'd already know by the time i get to this section what the header length is. In my head somehow I return automatically after the number of reads equals the length plus one or something, but I'm not sure how that would work. Maybe this is the best way, it just didn't feel right.